### ISSUE TITLE
Sampling for online evaluation + clearer APIs

### ISSUE THREAD

**Issue by joshreini1** [truera/trulens#2630]
## Description

By default, trulens computes metrics immediately after each trace is logged — so evaluation is already online and continuous. The gap is on high-traffic production apps: evaluating every trace is costly, and there's no easy sampling path to evaluate only a fraction of logged traces, nor clear APIs for controlling online evaluation.

## Proposed Change

1. **Sampling config**: a sampling policy (`sample_rate`, optional filter, throttle, cost budget) so you can evaluate only a fraction of logged traces — e.g. score 10% of production traces to control cost — applied to the existing immediate-after-log metric computation.
2. **Clearer APIs**: a clear, explicit entry point for enabling/disabling online evaluation and configuring sampling (today the immediate-eval default is implicit).
3. **Sampling context on results**: evaluated results carry the sampling context (`sample_rate`, sampled flag) so coverage is visible in the dashboard.
4. **Guardrails**: throttling so a high-traffic app doesn't overwhelm eval; per-app sampling rates.
5. **Cost budget — only when costs are actually tracked.** `cost_budget` is only enforceable when the metrics' provider reports tracked costs (OpenAI, LiteLLM, Google, Cortex today; other providers may not report costs). If `cost_budget` is set but a configured metric's provider does not report tracked costs, emit an **obvious, prominent warning** naming the metric + provider (at `configure_online_eval` time and in the dashboard) so the user knows the budget won't be enforced for that metric — do not silently ignore it.
6. **Alerting integration**: sampled online results can trigger alerts (tie to #2619's alerting viz).

## Usage Example

```python
from trulens.core import TruSession, Metric
from trulens.providers.openai import OpenAI
from trulens.apps.app import TruApp

session = TruSession()
provider = OpenAI()

# Metrics — by default these are computed immediately after each trace is logged
f_groundedness = (
    Metric(provider.groundedness_measure_with_cot_reasons, name="Groundedness")
    .on_input()
    .on_output()
)
f_relevance = (
    Metric(provider.relevance_with_cot_reasons, name="Answer Relevance")
    .on_input()
    .on_output()
)

# Wrap the app. By default every invocation evaluates all metrics right after the trace logs.
tru_app = TruApp(
    my_rag_app,                         # your RAG / app object
    app_name="prod_rag",
    app_version="v1",
    feedbacks=[f_groundedness, f_relevance],
)

# On a high-traffic app, evaluate only a sample of traces to control cost.
session.configure_online_eval(
    sample_rate=0.1,        # evaluate ~10% of logged traces
    apps=["prod_rag"],       # scope to this app
    throttle=100,            # max evaluations per minute
    cost_budget=10.0,        # daily $ cap — only enforced for providers with tracked costs
)
# If any metric's provider doesn't report tracked costs, a prominent warning is emitted,
# e.g.: "cost_budget set but 'f_relevance' uses 'acme' which does not report costs;
#        budget will not be enforced for that metric." — so the gap is obvious.

# Normal usage — traces are logged and (sampled) metrics computed automatically
for q in production_queries:
    with tru_app as recording:
        answer = my_rag_app.invoke(q)

# Inspect evaluated results and their sampling context
records = session.get_records_and_feedback(app_name="prod_rag")
# evaluated rows carry the sampling context so coverage is visible
evaluated = records[records["sampled"] == True]   # ~10% of traces, with sample_rate
```

## Files to Reference
- `src/core/trulens/core/session.py` — online-eval + sampling configuration; cost-tracking check
- `src/feedback/trulens/feedback/computer.py` — `compute_metrics` (the immediate computation)
- `src/core/trulens/experimental/otel_tracing/core/session.py` — `_track_costs` (~line 171) shows which providers report tracked costs
- `src/core/trulens/experimental/otel_tracing/core/exporter/connector.py` — ingest callback point

## Acceptance Criteria
- Configure sampling at `sample_rate=0.5`; ~50% of logged traces get metrics computed and the rest are skipped; evaluated results carry the sampling context (`sample_rate`, sampled flag); throttle is respected; alerts fire on threshold breach.
- If `cost_budget` is set and any configured metric's provider does not report tracked costs, `configure_online_eval` emits a clear, prominent warning naming the metric + provider; the budget is enforced only for metrics with tracked costs.

## Dependencies
- Reuses `compute_metrics`. Pairs with (but does not hard-depend on) #2619's alerting viz.
